### DISCUSSION TITLE
Switching back to old CZERTAINLY version

### DISCUSSION THREAD

**Discussion by ghost** [OmniTrustILM/ilm#40] [Q&A] [answered]
Hello,
I'm currently testing some new funcionality which is available only in develop-latest version of CZERTAINLY. In the past I have done the same thing and after I have had finished testing I switched back to normal version of CZERTAINLY (not the development one), but then we couldn't use CZERTAINLY because the database did not handle the version change well and was unreachable due to it, se we had to completely reinstall CZERTAINLY.

Is it now somehow fixed? Or after installing develop-latest CZERTAINLY version it is still not possible to switch back to normal version if one wants to avoid CZERTAINLY inoperability?

Best regards,
Denys Doloban

**Comment by 3keyroman** [comment, marked as answer]
Hello @doloban , once you do the upgrade, there is no path going back to older versions of CZERTAINLY, nor its components.

In some cases, it might be compatible, especially when there is no change in the database model or migrations, however, you should not rely on that.

if you would like to test new features or test upgrade of CZERTAINLY, it is recommended to do the backup of the database as a first step.

When you have the database backup, you can easily restore the same version of CZERTAINLY you had before upgrade.

We do not have any plans to support switching back to older versions of CZERTAINLY.

**Comment by ghost** [comment]
Hello,
@lubomirwasserman @3keyroman, are you planning on adding 2.13 version into latest-develop version 0.0.0 some time soon? Since I'm using 0.0.0 version for CZERTAINLY it is required for upgrade to 2.13, otherwise we will have to restore snapshot with CZERTAINLY with normal version.
